tests/math/test_is_independent.py

```diff
@@ -0,0 +1,499 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :func:`pennylane.math._is_independent` function.
+"""
+import pytest
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.math import _is_independent
+from pennylane.math._is_independent import _get_random_args
+
+try:
+    import jax
+
+    have_jax = True
+except ImportError:
+    have_jax = False
+try:
+    import torch
+
+    have_torch = True
+except ImportError:
+    have_torch = False
+try:
+    import tensorflow as tf
+
+    have_tf = True
+except ImportError:
+    have_tf = False
```

Comment:
Perhaps may as well just do the following? 

https://github.com/PennyLaneAI/pennylane/blob/8a14365e206771874dcfb48ef6d68d51de9118ba/tests/math/test_functions.py#L26-L29

The test will only be run when all interfaces exist, but perhaps that is fine...

Reply:
This is more or less how I had it before. But then in the CI there are multiple environments that will not look at this test...

Reply:
That should be okay if you want to make the change, since there is always 1 environment with all autodiff frameworks available :)